jni/src/main/java/com/horizen/certnative/WithdrawalCertificate.java

```diff
@@ -83,6 +83,10 @@ public void setScId(FieldElement scId) {
         this.scId = scId;
     }
 
+    public void setScId(byte[] scId) {
+        this.scId = FieldElement.deserialize(scId);
```

Comment:
Be careful here, as you are replacing this.scId thus losing the reference to it. You are not able to free it anymore causing a memory leak. You need to free it before replacing

Reply:
Done